src/main/java/fr/openent/competences/controllers/CompetenceNoteController.java

```diff
@@ -358,33 +337,32 @@ public void getCompetenceNoteDomaineClasse(final HttpServerRequest request) {
      * @param idPeriode
      * @param request
      */
-    private void callCompetencesNotesService(List<String> idEleves, Long idPeriode, HttpServerRequest request) {
+    private void callCompetencesNotesService(List<String> idEleves, Long idPeriode, Boolean isSkillAverage, HttpServerRequest request) {
         if (null != idEleves
                 && !idEleves.isEmpty()) {
-            competencesNotesService.getCompetencesNotesClasse(idEleves, idPeriode, arrayResponseHandler(request));
-        }
-    }
+            competencesNotesService.getMaxOrAverageCompetencesNotesClasse(idEleves, idPeriode, isSkillAverage,
+                    event ->{
+                if(event .isLeft()) {
+                    JsonObject error = (new JsonObject()).put("error", (String)event.left().getValue());
```

Comment:
Tu as des parenthèses redondantes tu pourrais faire ça : 

```suggestion
    JsonObject error = new JsonObject().put("error", event.left().getValue());
```
Et gérer le magic string "error" si possible!